catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/VisualPlacementActivityTest.java

```diff
@@ -0,0 +1,129 @@
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.content.brick.app;
+
+import android.support.test.espresso.Espresso;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.bricks.PlaceAtBrick;
+import org.catrobat.catroid.test.utils.TestUtils;
+import org.catrobat.catroid.testsuites.annotations.Cat;
+import org.catrobat.catroid.testsuites.annotations.Level;
+import org.catrobat.catroid.ui.SpriteActivity;
+import org.catrobat.catroid.uiespresso.util.rules.FragmentActivityTestRule;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils.createProjectAndGetStartScript;
+
+@RunWith(AndroidJUnit4.class)
+public class VisualPlacementActivityTest {
+	private int brickPosition;
+	private static final int XPOS = 200;
+	private static final int YPOS = -500;
+
+	@Rule
+	public FragmentActivityTestRule<SpriteActivity> baseActivityTestRule = new
+			FragmentActivityTestRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION, SpriteActivity.FRAGMENT_SCRIPTS);
+
+	@After
+	public void tearDown() throws Exception {
+		TestUtils.deleteProjects(VisualPlacementActivityTest.class.getSimpleName());
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		brickPosition = 1;
+		PlaceAtBrick brick = new PlaceAtBrick();
+		brick.setCoordinates(XPOS, YPOS);
+		createProjectAndGetStartScript(PlaceAtBrickTest.class.getSimpleName())
+				.addBrick(brick);
+		baseActivityTestRule.launchActivity();
+	}
+	@Category({Cat.AppUi.class, Level.Smoke.class})
+	@Test
+	public void testNoChangeConfirm() {
+
+		onBrickAtPosition(brickPosition).performClick();
+		onView(withText(R.string.brick_place_at_option_place_visually))
+				.check(matches(isDisplayed()))
+				.perform(click());
```

Comment:
leaving the activity under test (as in the test rule) is problematic in espresso and can lead to very weird failures of other tests, timeouts, activities getting stuck, etc. this in turn will lead to *other* tests randomly failing, which can make those failures very hard to locate and fix.

the nicer way of doing that would be testing each activity in isolation.
so you could test the VisualPlacementActivity in standalone by just giving it the intent extra on launch and see that it works as expected, aswell as that the activity result is correct.

using espressos Intended and Intending you can test the script activity for the brick by capturing the intent and asserting that the contents are correct.